### ISSUE TITLE
Crash when change address holds immature coins

### ISSUE THREAD

**Issue by alecalve** [wizardsardine/liana#1033]
Using signet, I generated coins to a change address of my wallet.

Starting Liana, I was then greeted by this in the logs (the UI just crashed):

```
ERROR liana:56: panic occurred at line 451 of file /vendor/liana/src/database/sqlite/mod.rs: Some("Database must be available: SqliteFailure(Error { code: ConstraintViolation, extended_code: 275 }, Some(\"CHECK constraint failed: is_change IS 0 OR is_immature IS 0\"))")
   0: liana::setup_panic_hook::{{closure}}
   1: std::panicking::rust_panic_with_hook
   2: <unknown>
   3: std::sys_common::backtrace::__rust_end_short_backtrace
   4: rust_begin_unwind
   5: core::panicking::panic_fmt
   6: core::result::unwrap_failed
   7: <liana::database::sqlite::SqliteConn as liana::database::DatabaseConnection>::new_unspent_coins
   8: liana::bitcoin::poller::looper::updates
   9: liana::bitcoin::poller::looper::looper
  10: std::sys_common::backtrace::__rust_begin_short_backtrace
  11: core::ops::function::FnOnce::call_once{{vtable.shim}}
  12: <unknown>
  13: start_thread
             at ./nptl/pthread_create.c:442:8
  14: __GI___clone3
             at ./misc/../sysdeps/unix/sysv/linux/x86_64/clone3.S:81
```

Mining an extra 110 blocks to another address made the UI start.

**Comment by jp1ac4** [comment]
Thanks very much for reporting this crash.

We've recently encountered this issue and are fixing it as part of https://github.com/wizardsardine/liana/pull/1001. The fix will be to remove the DB constraint that prevents an immature coin being held at a change address.

**Comment by jp1ac4** [comment]
This has now been fixed by #1001 and will be included in the upcoming v5 release.
